pep-9999.rst

```diff
@@ -0,0 +1,109 @@
+PEP: 9999
+Title: Disabling bdist_egg distribution uploads on PyPI
+Author: William Woodruff <william@yossarian.net>
+Sponsor: Donald Stufft <donald@stufft.io>
+PEP-Delegate: Donald Stufft <donald@stufft.io>
+Status: Draft
+Type: Process
+Topic: Packaging
+Content-Type: text/x-rst
+Created: 06-Jun-2023
+Post-History:
+Resolution:
+
+Abstract
+========
+
+This PEP recommends deprecating and then disabling new uploads of the
+``bdist_egg`` distribution type on PyPI. In a parallel move, this PEP recommends
+deprecating and then disabling new uploads of distribution filenames that have
+the ``.egg`` suffix.
+
+It does not recommend removing or otherwise affecting any previously
+uploaded ``bdist_egg`` distributions.
+
+Rationale
+=========
+
+Previous Work
+-------------
+
+The groundwork for this proposal was established with :pep:`527`, which
+proposed deprecating and eventually removing upload support for a handful
+of un(der)used file extensions and distribution types.
+
+In particular, :pep:`527` proposed the removal of the ``bdist_dumb``,
+``bdist_rpm``, ``bdist_dmg``, ``bdist_msi``, and ``bdist_wininst`` distribution
+types, as well as the ``.tar``, ``.tar.bz2``, ``.tar.xz``, ``.tar.Z``,
+``.tgz``, and ``.tbz`` file extensions on distribution filenames.
+
+:pep:`527` was fully enacted with
+`PR #7529 <https://github.com/pypi/warehouse/pull/7529>`_ to Warehouse,
+which was merged on 13 April 2020.
+
+The ``bdist_egg`` format
+------------------------
+
+The ``bdist_egg`` filetype identifies distributions in the
+`Egg format <https://packaging.python.org/en/latest/glossary/#term-Egg>`_. The
+Egg format was introduced by setuptools in 2004 and is roughly equivalent
+in functionality to the
+`Wheel format <https://packaging.python.org/en/latest/glossary/#term-Wheel>`_
+introduced and standardized by :pep:`427` in 2012.
+
+Despite its longevity, the Egg format has had
+`limited adoption on PyPI <https://github.com/pypi/warehouse/issues/10653>`_.
+Some observations from that issue:
+
+* As of December 2021, ``bdist_egg`` uploads account for less than 1% of all
+  built distribution uploads;
+* ``pip`` deprecated its ``--egg`` option in 2016;
+* ``setuptools`` has considered Egg support deprecated since 2019;
+* ``build`` only supports the ``sdist`` and ``bdist_wheel`` filetypes.
+
+Given the above, this PEP proposes the removal of the ``bdist_egg`` format
+under the same justifications presented in :pep:`527`, namely:
+
+1. Egg distributions are of of limited use to the broader ecosystem and
+   therefore represent a non-reciprocal maintenance burden;
+2. Continued support for Egg distributions represents a bandwidth and storage
+   cost to PyPI, as well as to any external mirrors of PyPI.
```

Comment:
```suggestion
* Egg distributions are of of limited use to the broader ecosystem and
  therefore represent a non-reciprocal maintenance burden;
* Continued support for Egg distributions represents a bandwidth and storage
  cost to PyPI, as well as to any external mirrors of PyPI.
```

This list doesn't appear to have an explicit, fixed order (as in a sequence of steps), so you might want to make it unordered instead.